### DISCUSSION TITLE
Formatter not found until an LSP is loaded

### DISCUSSION THREAD

**Discussion by alewis001** [LazyVim/LazyVim#4596] [Q&A] [answered]
If I do the following:

1. Open nvim
2. Paste in some JSON (single line)
3. Do `:set filetype=json`
4. Do `<leader>cf`

I get a message to say "no formatter available".

If instead I do:
1. Open nvim in a typescrpt/vue project
2. Open a file from the project (LSP Loads)
3. Open a new empty buffer
4. Paste in the same json as I did above
5. Do `:set filetype=json`
6. Do `<leader>cf`

The JSON is formatted.

Note that I must open the empty buffer after an LSP has loaded before pasting in the json, setting the filetype, etc. otherwise it still says "no formatter available".

I've gone back a few updates (lazy-lock revisions) and run `:Lazy restore` each time but the behaviour appears to be the same. So before I try going back any further, I wanted to check whether my expectations were correct or not :) 

Thanks,
Alex

**Comment by dpetka2001** [comment, marked as answer]
LazyVim uses `conform.nvim` as its default formatter. `conform.nvim` has to be configured either for a specific formatter for a specific filetype or else it defaults back to LSP formatting. That's why you're seeing this message. Read the `conform.nvim` docs in its Github repo for how to configure it. LazyVim [docs](https://www.lazyvim.org/plugins/formatting) also have some information.

**Reply by alewis001** [reply]
Ah ok, I see. Thanks! I'll take a look at the docs.
